cli/command/cli.go

```diff
@@ -151,6 +157,51 @@ func (cli *DockerCli) Initialize(opts *cliflags.ClientOptions) error {
 	return nil
 }
 
+// NotaryClient provides a Notary Repository to interact with signed metadata for an image
+func (cli *DockerCli) NotaryClient(imgRefAndAuth trust.ImageRefAndAuth, actions []string) (notaryclient.Repository, error) {
+	return trust.GetNotaryRepository(cli.In(), cli.Out(), UserAgent(), imgRefAndAuth.RepoInfo(), imgRefAndAuth.AuthConfig(), actions...)
+}
+
+// GetImageReferencesAndAuth retrieves the necessary reference and auth information for an image name
+// as a ImageRefAndAuth struct
+func GetImageReferencesAndAuth(ctx context.Context, cli Cli, imgName string) (*trust.ImageRefAndAuth, error) {
```

Comment:
ah ok, thanks for clarifying!  This will cause a package import cycle because of the call to `command.ResolveAuthConfig`, unless we're ok with shifting that code around or duplicating it?

Reply:
Ah, right! I ran into that problem with the manifest command as well. I solved it like this (https://github.com/docker/cli/pull/138/files#diff-b4a7ebaa86bd3765289ff28c6934850fR122):

```go
func (cli *DockerCli) RegistryClient() registryclient.RegistryClient {
 	resolver := func(ctx context.Context, index *registrytypes.IndexInfo) types.AuthConfig {
 		return ResolveAuthConfig(ctx, cli, index)
 	}
 	return registryclient.NewRegistryClient(resolver, UserAgent())
 }
```

`GetImageReferencecsAndAuth()` could accept a `resolver func(ctx context.Context, index *registrytypes.IndexInfo)` as an arg instead of the `cli Cli` parameter.

Reply:
ok this worked great 👍